build.gradle.kts

```diff
@@ -1,214 +1,259 @@
+    attributes["Build-Date"] = LocalDateTime.now()
+
+    val (branch, gitRevision, qualifier) = getSourceControlMetadata()
+    attributes["Full-Buildstring"] = "$branch@$gitRevision"
+    attributes["Git-Revision"] = gitRevision
+    attributes["Build-Qualifier"] = qualifier
+}
+
+// Main JAR
+tasks.jar {
+    dependsOn(configurations.runtimeClasspath)
+    from(sourceSets.main.get().output)
+    from({
+        configurations.runtimeClasspath.get().filter { it.name.contains("javahelp") && it.name.endsWith("jar") }.map { zipTree(it) }
+        configurations.runtimeClasspath.get().filter { it.name.contains("picocli") && it.name.endsWith("jar") }.map { zipTree(it) }
+    })
+    manifest {
+        attributes["Main-Class"] = application.mainClassName
+        from(sharedManifest)
+    }
+    dependsOn("copyHelp")
+}
+
+tasks.assemble{
+    dependsOn("jar") 
+}
+
+// NoHelp JAR
+tasks.create<Jar>("noHelpJar"){
+    classifier = "nohelp"
+    dependsOn(configurations.runtimeClasspath)
+    from(sourceSets.main.get().output)
+    from({
+        configurations.runtimeClasspath.get().filter { it.name.contains("picocli") && it.name.endsWith("jar") }.map { zipTree(it) }
+    })
+    manifest {
+        attributes["Main-Class"] = application.mainClassName
+        from(sharedManifest)
+    }
+}
+
+/*
+ * Code coverage report tasks
+ */
+tasks.jacocoTestReport {
+    reports {
+        xml.isEnabled = true
+        csv.isEnabled = false
+        html.isEnabled = false
+    }
+}
+
+tasks.check{
+    dependsOn("jacocoTestReport")
+}
+
+tasks.register("release") {
+    group = "Release"
+    description = "Creates all artifacts for a given EduMIPS64 release"
+    dependsOn("allDocs")
+    dependsOn("jar")
+
+    doFirst {
+        println("Creating artifacts for version $version")
+    }
+}
+
+tasks.create<Exec>("exeCreator"){
+    group = "Distribution"
+    description = "Creates an installable exe file"
+    workingDir = File("${projectDir}")
+    dependsOn("jar")
+
+    doFirst{
+    	//check if user is on Windows
+        var os = System.getProperty("os.name") as String;
+        if(!("Windows" in os)){
+            throw GradleException("This tasks shall run on Windows")
+        }
+
+        println("Creating edumips64-"+version+".jar File");
+        
+        //check if jdk 14+ is installed
+        if(System.getProperty("java.version").toInt()<14) {
+            throw GradleException("JDK version installed must be 14+")
+        }
+        else{
+            println("Correct Jdk version Installed!")
+        }
+        
+        //check if Wix is installed
+        if(System.getenv("WIX")==null){
+            throw GradleException("Please Install Wix")
+        }
+        else{
+            println("Wix Installed!")
+        }
+
+	//create script file that creates the executable
+        var script = File("./scripts/exeCreator.bat")
+        script.createNewFile()
+        var version_arr = version.split('.').toMutableList();

```

Comment:
Is this code necessary to truncate the version number? If so, could you please add a comment and emit a warning to the user that the version number is truncated?

Reply:
I followed the pattern on the jpackage command you sent me.
e.g. If version is 1.2.7.1, jpackage version is 1.2.8, jpackage doesn't allow the first version model.
The separation in a list is mandatory to allow the last number to have more than 1 digit

Reply:
1.2.7.1 is a very special case, we can assume I'll never use the 4th number and simplify the code so that it just uses the version number.

I'll bump up the version to 1.2.8 as soon as possible.

It's OK if you just add a TODO comment and I can simplify it once I bump up the version to 1.2.8.

Reply:
I filed https://github.com/EduMIPS64/edumips64/issues/465 to keep track of this clean-up, looks good as it is!